src/App.jsx

```diff
@@ -1,28 +1,141 @@
+/* eslint-disable react/no-access-state-in-setstate */
 import React from 'react';
 import './App.scss';
 import './styles/general.scss';
+import classNames from 'classnames';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api/api';
 
 class App extends React.Component {
   state = {
     todos: [],
+    copiedTodos: [],
     selectedUserId: 0,
+    title: '',
+    selectOption: '',
+    randomOption: false,
   };
 
+  componentDidMount() {
+    this.updateTodos();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.title !== this.state.title) {
+      this.titleParse();
+    }
+
+    if (prevState.randomOption !== this.state.randomOption) {
+      this.randomOrder();
+    }
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+
+    this.setState({ [name]: value });
+  }
+
+  updateTodos = async() => {
+    const todos = await getTodos();
+
+    this.setState({
+      todos,
+      copiedTodos: [...todos],
+    });
+  }
+
+  selectUserId = (id) => {
+    this.setState({ selectedUserId: id });
+  }
+
+  randomOrder() {
+    const newArr = (this.state.randomOption)
+      ? this.state.copiedTodos.sort(() => Math.random() - 0.5)
+      : this.titleParse();
+
+    this.setState({ copiedTodos: newArr });
+  }
+
+  titleParse() {
+    const newTodos = this.state.todos.filter((todo) => {
+      if (todo.title === null) {
+        return false;
+      }
+
+      return todo.title.includes(this.state.title);
+    });
+
+    this.setState({ copiedTodos: newTodos });
+
+    return newTodos;
+  }
+
   render() {
-    const { todos, selectedUserId } = this.state;
+    const { selectedUserId, title, copiedTodos, selectOption, randomOption }
```

Comment:
generate there array to show on ui, create a method `getVisibleTodos` in which you make filtering, and result use there, no copies in the state